### Easy Analytics (before breakfast ;-)
Wir laden unseren finalen Datensatz und legen los...

In [1]:
import pandas as pd
from fastparquet import ParquetFile

In [2]:
# Load Data
pf = ParquetFile('NYTaxi-TripData/new_york_taxi_cleaned_final.parquet')
df = pf.to_pandas()
df.head()

VendorID tpep_pickup_datetime tpep_dropoff_datetime  passenger_count  \
0       1.0  2019-11-01 00:35:48   2019-11-01 01:13:12              1.0   
1       1.0  2019-11-01 00:02:59   2019-11-01 00:05:26              1.0   
2       1.0  2019-11-01 00:11:43   2019-11-01 00:19:43              1.0   
3       1.0  2019-11-01 00:11:34   2019-11-01 00:18:02              1.0   
4       1.0  2019-11-01 00:54:04   2019-11-01 01:05:24              2.0   

   trip_distance  RatecodeID store_and_fwd_flag  payment_type  fare_amount  \
0            8.5         1.0                  N           1.0         32.0   
1            0.3         1.0                  N           2.0          3.5   
2            1.9         1.0                  N           2.0          8.5   
3            0.5         1.0                  N           2.0          5.5   
4            2.4         1.0                  N           1.0         10.0   

   extra  ...  total_amount  congestion_surcharge  LocationID_x  PU_Borough  \
0    3.0  ...         50.27                   2.5           138      Queens   
1    3.0  ...          7.30                   2.5           239   Manhattan   
2    3.0  ...         12.30                   2.5           142   Manhattan   
3    3.0  ...          9.30                   2.5           246   Manhattan   
4    3.0  ...         16.55                   2.5           246   Manhattan   

                     PU_Zone  service_zone_x  LocationID_y DO_Borough  \
0          LaGuardia Airport        Airports           161  Manhattan   
1      Upper West Side South     Yellow Zone           142  Manhattan   
2        Lincoln Square East     Yellow Zone           262  Manhattan   
3  West Chelsea/Hudson Yards     Yellow Zone           186  Manhattan   
4  West Chelsea/Hudson Yards     Yellow Zone           239  Manhattan   

                        DO_Zone service_zone_y  
0                Midtown Center    Yellow Zone  
1           Lincoln Square East    Yellow Zone  
2                Yorkville East    Yellow Zone  
3  Penn Station/Madison Sq West    Yellow Zone  
4         Upper West Side South    Yellow Zone  

[5 rows x 24 columns]

---

1. #### Wie viele Fahrten gibt es insgesamt?
* Aufgabe: Finde die Anzahl der Zeilen (Fahrten) im DataFrame.

In [3]:
df.shape[0]

99574711

#### 2. Wie viele Spalten gibt es im Datensatz?
* Aufgabe: Finde die Anzahl der Spalten im DataFrame.

In [4]:
df.shape[1]

24

#### 3. Wie viele einzigartige Anbieter (VendorID) gibt es?
* Aufgabe: Zähle, wie viele verschiedene Taxi-Anbieter im Datensatz vorkommen.

In [5]:
df['VendorID'].nunique()

3

#### 4. Was ist die maximale Anzahl von Passagieren in einer Fahrt?
* Aufgabe: Finde den höchsten Wert in der Spalte passenger_count.

In [6]:
df['passenger_count'].max()

np.float64(9.0)

#### 5. Was ist der niedrigste Fahrpreis (fare_amount)?
* Aufgabe: Finde den niedrigsten Wert in der Spalte fare_amount.
* Aufgabe: Finde die Anzahl der Fahrten mit einem Fahrpreis unter $2.50
* Aufgabe: Gib das Ergebnis in Prozent aus

In [7]:
df['fare_amount'].min()

np.float64(0.01)

In [10]:
# Anzahl der Fahrten mit Fahrpreis unter 2,5 ermitteln
anzahl_niedrige_preise = df[df['fare_amount'] < 2.5].shape[0]

print(f"Es gibt {anzahl_niedrige_preise} Fahrten mit einem Fahrpreis unter 2,5 $")

# Optional: Prozentsatz dieser Fahrten an der Gesamtmenge
prozentsatz = (anzahl_niedrige_preise / df.shape[0]) * 100
print(f"Das entspricht {prozentsatz:.2f}% aller Fahrten im Datensatz")


Es gibt 5458 Fahrten mit einem Fahrpreis unter 2,5 $
Das entspricht 0.01% aller Fahrten im Datensatz


#### 6. Was ist der durchschnittliche Trinkgeldbetrag (tip_amount)?
* Aufgabe: Berechne den Mittelwert der Trinkgelder.

In [11]:
df['tip_amount'].mean()

np.float64(2.1942434305433234)